app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+/**
+ * This view is the drawable area of the app.
+ * drawing on this location will draw to a bitmap which is loaded
+ * this saves the paths that are drawn on it
+ */
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    private var userDrawnBitmap: Bitmap? = null
+    private var intermediaryCanvas: Canvas? = null
+
+    var pathPaint = Paint().apply {
+        color = Color.BLACK
+        style = Paint.Style.STROKE
+        strokeWidth = 15f
+        strokeJoin = Paint.Join.ROUND
+        strokeCap = Paint.Cap.ROUND
+    }
+    private var path: Path? = null
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    path = Path()
+                    path?.moveTo(mEvent.x, mEvent.y)
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    path?.lineTo(mEvent.x, mEvent.y)
+                    intermediaryCanvas = userDrawnBitmap?.let { Canvas(it) }
+                    path?.let { path -> intermediaryCanvas?.drawPath(path, pathPaint) }
+                    invalidate()
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                    path?.lineTo(mEvent.x, mEvent.y)
```

Comment:
Don't you want to draw to the bitmap on up or cancel?

Reply:
Asked in dms to ensure I understood the reasoning for this correctly, added it in to draw to bitmap on up or cancel.